src/rrt/Tree.hpp

```diff
@@ -23,13 +24,31 @@ namespace RRT {
 template <typename T>
 class Node {
 public:
-    Node(const T& state, Node<T>* parent = nullptr)
-        : _parent(parent), _state(state) {
+    Node(const T& state, Node<T>* parent = nullptr, int dimensions = 2,
+         std::function<void(T, double*)> TToArray = NULL)
+        : _parent(parent), _state(state), _vec(dimensions) {
         if (_parent) {
             _parent->_children.push_back(this);
         }
+        _coordinates = new double[dimensions];
```

Comment:
It would really be best for this not be a stack allocaated raw pointer. Is that necessary? I think it can be just a vector right?

Best would be to change dimensions to a template parameter(It seems like it should be known at compile time) and have a std::array as a class variable. Alternatively a std::vector should also be a simpler replacement.